v1/tester/runner.go

```diff
@@ -491,28 +505,56 @@ func (r *Runner) runTests(ctx context.Context, txn storage.Transaction, enablePr
 		}
 	}
 
-	filenames := util.KeysSorted(r.compiler.Modules)
+	return testRegex, nil
+}
+
+func (r *Runner) runTests(ctx context.Context, txn storage.Transaction, enablePrintStatements bool, runFunc run) (chan *Result, error) {
```

Comment:
I just realized that originally the results sent to the channel returned by `runner.RunTests` were expected to be sorted by filename already. Now the sorting has to happen afterwards. I could change this and move the sorting to be done when the filter trace is added OR add a new `RunTests` methods that returns the results already grouped by filename. Then existing uses of `RunTests` won't break.

I am leaning towards adding a new method, I originally thought I could avoid it. Would also be make streaming the results easier if they were already grouped by filename. Then it doesn't have to wait to collect all the results to make sure each file is done.

Reply:
Resolved by collecting the results before sending them to ch, no noticeable slowdown and don't have to add new methods. Regal can then adopt this change with just a version bump.

Updated the test `TestRunnerPrintOutput` to check that the tests are grouped by file as the reporters expect.